Harvest.Web/ClientApp/src/Requests/RequestContainer.tsx

```diff
@@ -58,6 +59,18 @@ export const RequestContainer = () => {
     }
   }, [projectId]);
 
+  // This checks if the required fields are empty or not
+  // If they are undefined the submit button should be disabled
+  useEffect(() => {
+    if (
+      project.start !== undefined &&
+      project.end !== undefined &&
+      project.crop !== undefined
+    ) {
+      setButtonDisabled(false);
+    }
+  }, [project.crop, project.start, project.end]);
+
```

Comment:
Try this instead -- get rid of the state/boolean.  Change this to `useMemo`.  So 
```
var isFilledIn =  useMemo(() => {
    return
      project.start !== undefined &&
      project.end !== undefined &&
      project.crop !== undefined;
  }, [project.crop, project.start, project.end]);
```

then you just need to set button `disabled={!isFilledIn}`.  Save yourself some state and use less code for it.

Reply:
Okay I can update that. Also, when is it appropriate to use `useMemo` instead of `useEffect` in React?

Reply:
UseMemo returns a value so it's useful when you want the result of some calculation locally. UseEffect is just something that runs based on changed params, so good for making network calls or updating totals or variables.

Reply:
Ahh I see. So useMemo is mainly used when we don't want to cause side effects like making API calls while useEffect does with side effects?

Reply:
Yea that's how I tend to use them anyway. I suppose you could have a side effect inside memo but that's probably bad form.